internal/r8/deobfuscator.go

```diff
@@ -0,0 +1,205 @@
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// Checks if the found methodMatch from the map file is part of the methods previously parsed for the currentType within the
+// stacktrace.
+// If the method is found, then it's added to the mapping in order to be used for replacing the obfuscated name later, it also returns it as the current MappedMethodCall.
+// If the method is NOT found, but it turns out to be a continuation of a previously found method (currentMappedMethodCall), then it appends it to the existing method replacement in the mapping.
+// If the method is NOT found and is also NOT a continuation for the currentMappedMethodCall, then the methodMatch is ignored.
+func upsertMappedMethodCall(mapping map[string]string, methodMatch MethodMatch, currentType *MappedType, currentMappedMethodCall *MappedMethodCall) *MappedMethodCall {
+	methodNameReference := methodMatch.methodObfuscatedName
+	if methodMatch.sourceFileStart != "" {
+		if methodMatch.sourceFileStart != methodMatch.sourceFileEnd {
+			// This is probably due an edge-case where the mapping line starts with different numbers (e.g 1:2). We don't
+			// have that case in our tests, therefore we are ignoring it.
+			return currentMappedMethodCall
+		}
+		methodNameReference = fmt.Sprintf("%s:%s", methodMatch.methodObfuscatedName, methodMatch.sourceFileStart)
+	}
+	mapReference := currentType.obfuscated.name + ":" + methodNameReference
+	methodCallSite, ok := currentType.obfuscated.methods[methodNameReference]
+	if ok {
+		// Found this method in the list of methods parsed from the stacktrace for the currentType.
+		delete(currentType.obfuscated.methods, methodNameReference)
+		key := getKey(currentType.obfuscated.name, methodMatch.methodObfuscatedName, methodCallSite)
+		mapping[key] = getKey(currentType.realName, methodMatch.methodRealName, methodCallSite)
+		return &MappedMethodCall{mapReference, key}
+	} else if currentMappedMethodCall != nil && currentMappedMethodCall.reference == mapReference {
+		// This is a continuation call for the currentMappedMethodCall.
+		mapping[currentMappedMethodCall.key] += "\n" + fmt.Sprintf("%s%s", strings.Repeat(" ", len(currentType.realName)+currentType.obfuscated.indentation), methodMatch.methodRealName)
+	}
+
+	return currentMappedMethodCall
+}
+
+// Sometimes the map file contains a type but not a method name belonging to said type, this might happen because the type name was obfuscated
+// but its method wasn't, in this case we use this function to make sure we add the leftover de-obfuscated type names from the map file,
+// along with the method names as found in the stacktrace (which should be de-obfuscated already).
+func mapLeftoverUnmappedMethods(res map[string]string, currentType *MappedType) {
+	for methodName, callSite := range currentType.obfuscated.methods {
+		key := getKey(currentType.obfuscated.name, methodName, callSite)
+		res[key] = getKey(currentType.realName, methodName, callSite)
+	}
+}
```

Comment:
Sorry I missed this `res` earlier
```suggestion
func mapLeftoverUnmappedMethods(mapping map[string]string, currentType *MappedType) {
	for methodName, callSite := range currentType.obfuscated.methods {
		key := getKey(currentType.obfuscated.name, methodName, callSite)
		mapping[key] = getKey(currentType.realName, methodName, callSite)
	}
}
```